pkg/app/app_factory.go

```diff
@@ -21,6 +21,11 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+// ICRDAppFactory interface for CRDAppFactory
+type ICRDAppFactory interface {
```

Comment:
`ICRDAppFactory` Maybe not a good name let me know if you have some better suggestions.